checkbox-provider-dss/units/jobs.pxu

```diff
@@ -262,3 +262,232 @@ command:
+    echo "DSS has started successfully!"
+  }
+  remove_ipex() {
+    echo "Removing IPEX"
+    sudo microk8s.kubectl delete -n dss deployment.apps/notebook-pytorch
+    sudo microk8s.kubectl delete -n dss service/user-notebook-pytorch
+  }
+  launch_notebook pytorch ipex
+  echo "Starting ipex import test"
+  kubectl get pods -n dss -o=jsonpath='{.items..metadata.name}'
+  pod=$(kubectl get pods -n dss -o=jsonpath='{.items..metadata.name}' | grep -o 'notebook-pytorch.*')
+  echo "Found PyTorch pod: ${pod}"
+  gpu_grep_out=$(sudo microk8s kubectl -n dss exec ${pod} -- python3 -c '
+  import sys
+  import torch
+  import intel_extension_for_pytorch as ipex
+  print(torch.__version__)
+  print(ipex.__version__)
+  try:
+    [print(f"[{i}]: {torch.xpu.get_device_properties(i)}") for i in range(torch.xpu.device_count())];
+    sys.exit(0)
+  except Exception:
+    print("Encountered an error getting XPU device properties")
+    sys.exit(1)
+  ' | grep "dev_type=.gpu" 2>&1)
+  remove_ipex
+  if [[ -z ${gpu_grep_out} ]]; then
+    echo "FAIL: No GPU found"
+    exit 1
+  elif [[ $gpu_grep_out =~ "gpu" ]]; then
+    echo "PASS: GPU found"
+    exit 0
+  else
+    echo "FAIL: Unexpected result"
+    echo ${gpu_grep_out}
+    exit 1
+  fi
+
+id: itex/itex_gpu_avail
+category_id: dss-regress
+flags: simple
+requires: executable.name == 'microk8s'
+_summary: Check ITEX GPU Availability
+estimated_duration: 5m
+command:
+  TIMEOUT=1200s
+  DSS_INTEL_MANIFESTS=/tmp/data-science-stack/poc/intel/manifests/
+  launch_notebook() {
+    local fullname=$1
+    local app=$2
+    echo -e "\n\$ dss start ${app} notebook"
+    sudo microk8s kubectl apply -f ${DSS_INTEL_MANIFESTS}/notebook-${app}.yaml
+    sudo microk8s.kubectl wait \
+      --for condition=available \
+      --timeout $TIMEOUT \
+      -n dss\
+      deployment \
+      -l app=user-notebook-${fullname}
+    echo -e "\n\$ dss status"
+    MLFLOW_IP=$(sudo microk8s.kubectl get svc \
+      -n dss \
+      -o jsonpath="{.spec.clusterIP}" \
+      mlflow)
+    NOTEBOOK_IP=$(sudo microk8s.kubectl get svc \
+      -n dss \
+      -o jsonpath="{.spec.clusterIP}" \
+      user-notebook-${fullname})
+    echo "DSS has started successfully!"
+  }
+  remove_itex() {
+    echo "Removing ITEX"
+    sudo microk8s.kubectl delete -n dss deployment.apps/notebook-tensorflow
+    sudo microk8s.kubectl delete -n dss service/user-notebook-tensorflow
+  }
+  launch_notebook tensorflow itex
+  echo "Starting itex GPU check test"
+  pod=$(kubectl get pods -n dss -o=jsonpath='{.items..metadata.name}' | grep -o 'notebook-tensorflow.*')
+  echo "Found Tensorflow pod: ${pod}"
+  gpu_grep_out=$(sudo microk8s kubectl -n dss exec ${pod} -- python3 -c '
+  import tensorflow as tf
+  ' 2>&1 >/dev/null | grep "Intel Extension for Tensorflow\* GPU backend is loaded")
+  remove_itex
+  if [[ -z ${gpu_grep_out} ]]; then
+    echo "ERROR: No GPU found"
+    exit 1
+  elif [[ $gpu_grep_out =~ "GPU backend is loaded" ]]; then
```

Comment:
I think this line may get printed even without a GPU. When I was fighting with DSS today I saw the following output when a GPU was not detected inside the container:

```
2024-06-27 21:01:37.537970: I tensorflow/core/util/port.cc:110] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2024-06-27 21:01:37.561386: I tensorflow/tsl/cuda/cudart_stub.cc:28] Could not find cuda drivers on your machine, GPU will not be used.
2024-06-27 21:01:37.736859: I tensorflow/tsl/cuda/cudart_stub.cc:28] Could not find cuda drivers on your machine, GPU will not be used.
2024-06-27 21:01:37.739460: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-06-27 21:01:38.321887: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
2024-06-27 21:01:40.267786: I itex/core/wrapper/itex_gpu_wrapper.cc:35] Intel Extension for Tensorflow* GPU backend is loaded.
2024-06-27 21:01:40.368135: I itex/core/wrapper/itex_cpu_wrapper.cc:52] Intel Extension for Tensorflow* AVX2 CPU backend is loaded.
2024-06-27 21:01:40.416022: W itex/core/ops/op_init.cc:58] Op: _QuantizedMaxPool3D is already registered in Tensorflow
2024-06-27 21:01:40.426377: E itex/core/devices/gpu/itex_gpu_runtime.cc:173] Can not found any devices. To check runtime environment on your host, please run itex/tools/env_check.sh.
If you need help, create an issue at https://github.com/intel/intel-extension-for-tensorflow/issues
```

Reply:
OK, glad you found that. I just removed that bit so we aren't depending on that output